tests/installation/release_notes_from_url.pm

```diff
@@ -19,7 +19,15 @@ sub run {
     enter_cmd "zgrep -oh \"Got release notes.*\" /var/log/YaST2/y2log*";
     assert_screen [qw(got-releasenotes-RPM got-releasenotes-URL)];
     unless (match_has_tag 'got-releasenotes-URL') {
-        record_soft_failure('bsc#1190711 - Release notes source does NOT match expectations or not found in YaST logs, expected source: URL');
+        my $not_RC = script_run('grep -r RC /README.BETA');
```

Comment:
```suggestion
        my $error_msg = 'Release notes source does NOT match expectations or not found in YaST logs, expected source: URL';
    if (!match_has_tag 'got-releasenotes-URL') {
        # detect whether is public/publicRC
        my $is_rc = script_run('!grep -r RC /README.BETA');
        
        # Beta is dropped in GMC
        die $error_msg if !($is_rc || !get_var('BETA'));
        record_info('bsc#1190711', $error_msg);
    }
```

Reply:
- original `unless` cannot be reused because the code grow
- grep can be inverted, but reading command line is more obscure than `!`
- error variable is move up (I will explain why in next suggestion)
- we need comments for sure when there is something that is not clear by the code, the GMC one for sure, the first one can be replaced by naming the variable `my $is_rc_or_public_rc` but not fan of `and`s and `or`s in variable names, but in testing are common and helpful.
- with the `die` first unfortunately we need to reverse the logic but we avoid that `if/else` and it is habit to give the dies first in Perl for that reason.

Reply:
we would need to this code to be used for other bugs, could you find some library where to put this condition, name of the function, I don't know, `is_rc_or_public_rc_or_gmc` if you don't find any better for example.

Reply:
we need to update commit and PR title according to new code btw.

Reply:
There are 4 ! in your proposal and two of them are in the same condition, this is way more confusing than the code in the PR. If you are so much against the not_RC, I can use the ! to make it is_RC, inside some library (the most suitable I found is y2_installbase, to avoid the libyui libs that we don't want to add testapi etc) , but I don't think anything like line 28 would be helpful. Also, the is_GMC I think helps make the code cleaner, a comment might be helpful to clarify. Finally, why shouldn't I use unless with a bigger block of code? I just checked Perl tutorials out of curiosity and they have textbook examples of unless followed by big blocks. I could use "if (match_has_tag 'got-releasenotes-RPM') but I think unless works well here.

Reply:
the code looks great now, good job!
The first unless had a simple condition so it was ok if you want to keep it (I guess I suggested to not encourage more `unless` when they are complex).
Positive naming for variables is a must to understand code, thanks for taking a look to that.
Regarding the is_GMC, in Perl it is always better to not declare variable as much as possible, but sort of make sense to understand the code, so worth it with the comment.